backend/src/main/java/ca/bc/gov/hlth/mohums/controller/ClientController.java

```diff
@@ -0,0 +1,80 @@
+package ca.bc.gov.hlth.mohums.controller;
+
+import ca.bc.gov.hlth.mohums.webclient.WebClientService;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+@RestController
+public class ClientController {
+
+    private final WebClientService webClientService;
+    private final Logger logger = LoggerFactory.getLogger(ClientController.class);
+
+    private static final JSONParser jsonParser = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE);
+
+    public ClientController(WebClientService webClientService) {
+        this.webClientService = webClientService;
+    }
+
+    @GetMapping("/clients")
+    public Mono<Object> clients(@RequestHeader("Authorization") String token) {
```

Comment:
This class might be a good place to do some unit testing. It's behaviour is encapsulated and has no tricky dependencies. You pass in a base-64 encoded token, which you can build on the fly, and assert the expected values.

You would need to figure out how to mock `WebClientService`, which I have some ideas for if you like, or you could just factor this bit of code into a new class:

```java
JSONArray roles = parseRoles(token);
List<String> authorizedClients = parseClients(roles);
```

A class that takes a token and returns the authorized clients.

Anyhow, this is just a suggestion. I know we have a lot to learn and a lot to work on,and I don't know what the priority is. Trevor's integration tests will hopefully cover this anyhow, though I would like to also figure out unit testing...